docs/remote/devcontainer-cli.md

```diff
@@ -126,13 +126,87 @@ These steps above are also provided in the CLI repo's [README](https://github.co
 
 If you'd like to use the dev container CLI in your CI/CD builds or test automation, you can find examples of GitHub Actions and Azure DevOps Tasks in the [devcontainers/ci](https://github.com/devcontainers/ci) repository.
 
+## Pre-building
+
+The `devcontainer build` command allows you to build quickly build dev container image following the same steps the Remote - Containers extension or GitHub Codespaces will. This is particularly useful when you want to pre-build a dev container image using a CI or DevOps product like GitHub Actions.
+
+`build` accepts a path to the folder containing a `.devcontainer` folder or `.devcontainer.json` file. For example, `devcontainer build --workspace-folder <my_repo>` will build the container image for `my_repo`.
+
+### Example of building and publishing an image
+
+For example, you may want to pre-build a number of images that you then reuse across multiple projects or repositories. To do so, follow these steps:
+
+1. [Create](/docs/editor/versioncontrol.md#initialize-a-repository) a source code repository.
+
+1. Create dev container configuration for each image you want to pre-build, customizing as you wish (including [dev container Features](#dev-container-features-preview)). For example, consider this devcontainer.json file:
+
+    ```json
+    {
+        "build": {
+            "dockerfile": "Dockerfile"
+        },
+        "features": {
+            "docker-in-docker": "latest"
+        }
+    }
+    ```
+
+1. Use the `devcontainer build` command to build the image. See documentation for your image registry (like the [Azure Container Registry](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli), [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images), or [Docker Hub](https://docs.docker.com/engine/reference/commandline/push)) for information on image naming and additional steps like authentication.
+
+    ```bash
+    devcontainer build --workspace-folder <my_repo>
+    ```
+
+1. Next [push](https://docs.docker.com/engine/reference/commandline/push/) the image to your registry.
+
+    ```bash
+    docker push ghcr.io/your-org/your-image-name
+    ```
+
+1. Finally, for each project or repository that will use your image, craft a simplified devcontainer.json file that either uses the `image` property or references it in a Docker Compose file. Include any dev container features you added in your pre-build configuration in step 2. For example:
```

Comment:
FYI @chrmarti @alexdima - I raised https://github.com/devcontainers/spec/issues/74 to allow us to drop the need for a separate devcontainer.json.

An alternative near term would be to instead document the use of "cacheFrom" with a single devcontainer.json rather than highlight that a separate image could be used.  So we'd have:

```json
    {
        "build": {
            "dockerfile": "Dockerfile",
            "cacheFrom": "ghcr.io/your-org/your-image-name"
        },
        "features": {
            "ghcr.io/devcontainers/features/docker-in-docker": {}
        }
    }
```

The devcontainer.json is more complicated, but you'd only need one.

Thoughts?

Reply:
I like the idea of mentioning the `cacheFrom` alternative. I took a pass at adding it as an option, so that users can choose either the separate simplified devcontainer.json or using `cacheFrom`. Let me know what you think.